RT_UPDATE/basepop_single.R

```diff
@@ -0,0 +1,613 @@
+  if(!is.null(Females_single)) {
+    
+    Females_single <- tibble(pop    = Females_single,
+                             year   = refDate + 1,
+                             age    = row_number() - 1,
+                             cohort = year - age - 1)
+    
+    
+  }
+  
+  # same for Males
+  if(!is.null(Males_single)) {
+    
+    Males_single <- tibble(pop    = Males_single,
+                           year   = refDate + 1,
+                           age    = row_number() - 1,
+                           cohort = year - age - 1)
+    
+  }
+  
+  # --- Retrieve WPP population if user did not provide one -----------------
+  # if user did not provide Population
+  if(is.null(Females_single)) {
+    
+    Females_single <- popAge1dt |>
+      filter(country_code == !!country_code, year == refDate) |>
+      select(year, age, pop = popF) |>
+      mutate(year   = year + 1,
+             cohort = year - age - 1)
+    
+  }
+  
+  # same operation for males
+  if(is.null(Males_single)) {
+    
+    Males_single <- popAge1dt |>
+      filter(country_code == !!country_code, year == refDate) |>
+      select(year, age, pop = popM) |>
+      mutate(year   = year + 1,
+             cohort = year - age - 1)
+    
+  }
+  
+  # --- Age validation and setup --------------------------------------------
+  # Age setup
+  if(!is.null(Age)) {
+    
+    stopifnot(is_single(Age))
+    stopifnot(length(Age) == nrow(Females_single))
+    
+  } else {
+    
+    Age <- as.integer(sort(unique(Females_single$age)))
+    
+  }
+  
+  # --- Determine life table radix ------------------------------------------
+  if(is.null(radix) & !is.null(nLxFemale)) {
+    
+    radix <- lt_infer_radix_from_1L0(nLxFemale[1, 1])
+    
+    if(verbose) {
+      
+      cat(paste0("Setting radix to value of lx: ", radix, 
+                 ". Can be overwritten with the `radix` argument"),
+          sep = "\n")
+      
+    }
+    
+  }
+  
+  
+  if(is.null(radix) & is.null(nLxFemale)) { 
+    
+    radix <- 1
+    cat(paste0("Setting radix to value of lx: ", radix, 
+               ". Can be overwritten with the `radix` argument"),
+        sep = "\n")
+    
+  }
+  
+  # --- Compute reverse survival ratios (SxRev) ------------------------------
+  # These represent the probability of surviving backward in time.
+  # Female and male SxRev matrices are created either from nLx input or WPP mx data.
+  # Inflation factor is cumulative survival ratio (used for exposure reconstruction).
+
+  if(!is.null(nLxFemale)) {
+    
+    mxF <- nLxFemale %>% 
+      group_by(year) %>% 
+      mutate(n  = age2int(.data$age, OAvalue = Inf),
+             ax = ifelse(age == 0, 
+                         solve_a0_from_L0(L0_target = .data$Lxp[.data$age == 0],
+                                          l0        = radix,
+                                          sex       = "f"), 
+                         0.5)) %>% 
+      unnest(ax) %>% 
+      mutate(qx      = (Lxp - lead(Lxp)) / Lxp,
```

Comment:
this doesn't seem like a valid calc. treating Lx as if it were lx.
For age 0 you know l0 already and a0, then you can derive all age 0 quantities up until l1, then iterate in like fashion up the ages. Starting from Lx is awkward like this. Really it might make sense to outsource to a new helper, like lt_id_La_q() or lt_id_La_m(nLx, ax, radix (can be inferred)), and be sure to use DemoTools functions which are more robust rather than hard-coded lifetable operations. That all start with lt_id* i.e. lifetable identities.